### PR TITLE
Clean up relay module

### PR DESCRIPTION
**Pull request by philipktlin**
Clean up relay module; and add InputName to Settings.

### CODE REVIEW

File: edge-modules/Relayer/Settings.cs [hunk 1]
```diff
@@ -24,24 +26,40 @@ public class Settings
...
 
         Settings(
             TransportType transportType,
+            string inputName,
             string outputName)
         {
-            this.TransportType = Preconditions.CheckNotNull(transportType);
+            this.InputName = Preconditions.CheckNonWhiteSpace(inputName, nameof(inputName));
             this.OutputName = Preconditions.CheckNonWhiteSpace(outputName, nameof(outputName));
+            this.TransportType = transportType;
```
Comment (yophilav) [inline edge-modules/Relayer/Settings.cs]: Is there a Check if the enum is valid within a set of `TransportType`?

File: edge-modules/Relayer/Program.cs [hunk 2]
```diff
@@ -38,7 +40,7 @@ static async Task Main(string[] args)
             }
             catch (Exception ex)
             {
-                Logger.LogError($"Error occurred during Relayer.\r\n{ex}");
+                Logger.LogError(ex, $"Error occurred during Relayer.");
```
Comment (and-rewsmith) [inline edge-modules/Relayer/Program.cs]: We don't need the dollar sign anymore

File: edge-modules/Relayer/Settings.cs (hunk 1 above)
Comment (and-rewsmith) [reply edge-modules/Relayer/Settings.cs]: configuration.GetValue will throw an error if invalid

File: edge-modules/Relayer/Settings.cs [hunk 3]
```diff
@@ -24,24 +26,40 @@ public class Settings
...
+        {
+            // serializing in this pattern so that secrets don't accidentally get added anywhere in the future
+            var fields = new Dictionary<string, string>
+            {
+                { nameof(this.InputName), this.InputName },
+                { nameof(this.OutputName), this.OutputName },
+                { nameof(this.TransportType), Enum.GetName(typeof(TransportType), this.TransportType) },
+            };
+
+            return $"Settings:{Environment.NewLine}{string.Join(Environment.NewLine, fields.Select(f => $"{f.Key}={f.Value}"))}";
```
Comment (and-rewsmith) [inline edge-modules/Relayer/Settings.cs]: I prefer JsonConvert.SerializeObject as it is more readable, but your call here.

File: edge-modules/Relayer/Settings.cs (hunk 1 above)
Comment (philipktlin) [reply edge-modules/Relayer/Settings.cs]: Currently all values in TransportType enum are valid here.  Also afaik i don't know there is any method in Precondition class to verify enum.

File: edge-modules/Relayer/Program.cs (hunk 2 above)
Comment (philipktlin) [reply edge-modules/Relayer/Program.cs]:
```suggestion
                Logger.LogError(ex, "Error occurred during Relayer.");
```

File: edge-modules/Relayer/Settings.cs (hunk 3 above)
Comment (philipktlin) [reply edge-modules/Relayer/Settings.cs]: I just made it consistently with loadGen with same reason.

**Comment by and-rewsmith** [comment]
LGTM

**Pull request merged**